src/receive_imf.rs

```diff
@@ -2529,41 +2530,43 @@ async fn has_verified_encryption(
 ) -> Result<VerifiedEncryption> {
     use VerifiedEncryption::*;
 
-    // We do not need to check if we are verified with ourself.
-    let to_ids = to_ids
-        .iter()
-        .copied()
-        .filter(|id| *id != ContactId::SELF)
-        .collect::<Vec<ContactId>>();
-
     if !mimeparser.was_encrypted() {
```

Comment:
But they are already checked here in fact. `was_encrypted()` returns false if there are no valid signatures. Moreover, signatures are checked with all self keys (returned by `key::load_self_public_keyring()`) which looks more correct, i.e. all the self keys are verified